src/Aspire.Hosting/Pipelines/DistributedApplicationPipeline.cs

```diff
@@ -354,13 +354,16 @@ async Task ExecuteStepWithDependencies(PipelineStep step)
 
                         // Wrap the dependency failure with context about this step
                         var wrappedException = new InvalidOperationException(message, ex);
+                        step.Status = PipelineStepStatus.Failed;
```

Comment:
Do we need to worry about concurrency at all? Is it possible for 2 threads to try to update the status at the same time?